managed/devops/opscli/ybops/cloud/aws/cloud.py

```diff
@@ -381,11 +382,29 @@ def stop_instance(self, args):
         except ClientError as e:
             logging.error(e)
 
-    def start_instance(self, args):
+    def start_instance(self, args, ssh_port):
         ec2 = boto3.resource('ec2',  args["region"])
         try:
+            sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
             instance = ec2.Instance(id=args["id"])
             instance.start()
             instance.wait_until_running()
+            # The OS boot up may take some time,
+            # so retry until the instance allows SSH connection.
+            retry_count = 0
+            while retry_count < 30:
+                time.sleep(10)
+                retry_count = retry_count + 1
+                result = sock.connect_ex((args["private_ip"], ssh_port))
+                if result == 0:
+                    break
+            else:
+                logging.error("Start instance %s exceeded maxRetries!", args["id"])
+                raise YBOpsRuntimeError(
+                    "Can not start the instance %s as the instance is not reachable.", args["id"]
```

Comment:
For me a message like "Cannot reach the instance %s after its start." is better. Because actually you invoked the start but can't reach it after.
Also (!) include ssh port number into the message, sometimes it is very helpful.